lefthook.yml

```diff
@@ -0,0 +1,18 @@
+pre-commit:
+  parallel: true
+  commands:
+    frontend-lint:
+      tags: frontend lint
+      root: "frontend/"
+      glob: "*.{js,jsx,ts,tsx}"
+      run: eslint --fix {staged_files}
+    frontend-types:
+      tags: frontend types
+      root: "frontend/"
+      glob: "**/*.{ts,tsx}"
+      run: echo {staged_files} >/dev/null && yarn check-types
```

Comment:
lefthook is not without issues, like here {staged_files} must appear otherwise it will always do frontend types check even if there are no staged files matching `root` and `glob`. https://github.com/evilmartians/lefthook/issues/228#issuecomment-924505331

Reply:
Other solution is to use https://github.com/gustavopch/tsc-files instead of `tsc`.

Reply:
or moving type check to pre-push

Reply:
I think that'd be a good idea to move type checking to a pre-push hook instead.

I'm in favour of having every commit type checked but if this takes more than 1 or 2 seconds to execute, maybe we can only check when pushing to remote.